### PR TITLE
feat: add a blog post for CVE-2018-15685

### PR DESCRIPTION
**Pull request by ckerr**
Add a blog post for CVE-2018-15685.

### CODE REVIEW

File: data/blog/web-preferences-fix.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title:  WebPreferences Vulnerability Fix
+author: ckerr
+date: '2018-08-22'
+---
+
+A remote code execution vulnerability has been discovered affecting  apps with the ability to open consecutive child windows on Electron versions (3.0.0-beta.6, 2.0.7, 1.8.7, and 1.7.15) for. This vulnerability has been assigned the CVE identifier [CVE-2018-15685].
```
Comment (MarshallOfSound) [inline data/blog/web-preferences-fix.md]: s/consecutive/nested

(hierarchy is the important thing here)

Comment (zeke) [reply data/blog/web-preferences-fix.md]: The wording of "with the ability to open" is ambiguous -- does the CVE have this ability or does it affect apps that have this ability?

File: data/blog/web-preferences-fix.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+You are impacted if:
+
+1. You embed _any_ remote user content, even in a sandbox
+2. You accept user input with any XSS vulnerabilities
+
+_Details_
+
+You are impacted if any user code runs inside an `iframe` / can create an `iframe`. Given the possibility of an XSS it can be assumed that most apps are vulnerable to this case.
+
+You are also impacted if you open any of your windows with the `nativeWindowOpen: true` or `sandbox: true` option.  Although this vulnerability also requires an XSS to exist in your app, you should still apply one of the mitigations below if you use either of these options.
```
Comment (zeke) [inline data/blog/web-preferences-fix.md]: "an XSS" is not really a thing. Maybe "an XSS vulnerability" or "an XSS susceptibility" or similar.

File: data/blog/web-preferences-fix.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Mitigation
+
+We've published new versions of Electron which include fixes for  this vulnerability: [`3.0.0-beta.7`](https://github.com/electron/electron/releases/tag/v3.0.0-beta.7), [`2.0.8`](https://github.com/electron/electron/releases/tag/v2.0.8), [`1.8.8`](https://github.com/electron/electron/releases/tag/v1.8.8), and [`1.7.16`](https://github.com/electron/electron/releases/tag/v1.7.16). We urge all Electron developers to update their apps to the latest stable version immediately.
+
+If for some reason you are unable to upgrade your Electron version, you can protect your app by blanket-calling `event.preventDefault()` on the `new-window` event for all  `webContents`'. If you don't use `window.open` or any child windows at all then this is also a valid mitigation for your app.
+
+```javascript=
```
Comment (zeke) [inline data/blog/web-preferences-fix.md]: no need for `=` at the end of the code fencing.

File: data/blog/web-preferences-fix.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+mainWindow.webContents.on('new-window', e => e.preventDefault())
+```
+
+If you rely on the ability of your child windows to make grandchild windows, then a third mitigation strategy is to use the following code on your top level window:
+
+```javascript=
```
Comment (zeke) [inline data/blog/web-preferences-fix.md]: no need for `=` at the end of the code fencing.

**Pull request merged**